File: DataAccessLayer/Entities/Report.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataAccessLayer.Entities
{
    public class Report: BaseEntity
    {
        [StringLength(200)]
        public string Comment { get; set; }

        [Required]
        public int PhotoId { get; set; }

        [Required]
        public int UserId { get; set; }

        [Required]
        public ReportStatus Status { get; set; }

        public User User { get; set; }
        public Photo Photo { get; set; }

    }



    public enum ReportStatus
    {
        Approved,
        Declined,
        OnReview
    }
}

File: UI-Angular/src/app/models/report.ts
export class Report{
    id!: number;
    comment!: string;
    photoId: number;
    userId: number;
    status: ReportStatus;

    constructor(comment="", photoId:number,userId:number, status: ReportStatus ){
        this.comment = comment;
        this.photoId = photoId;
        this.userId = userId;
        this.status = status;
    }
}

export enum ReportStatus{
        Approved = 0,
        Declined = 1,
        OnReview = 2
}